events/api/src/main/java/org/projectnessie/events/api/TransplantEvent.java

```diff
@@ -30,4 +31,10 @@ public interface TransplantEvent
   default EventType getType() {
     return EventType.TRANSPLANT;
   }
+
+  /**
+   * The sequence of hashes from the source reference that were transplanted into the target
+   * reference.
+   */
+  List<String> getSourceHashes();
```

Comment:
Wanted to note: the initial design of this API was very cautious when adding lists of elements. E.g. the commit event does not contain a list of operations, but instead, each operation is emitted as a standalone event. This was donne to prevent giant commit events with thousands of operations.

Do we need to take this aspect into consideration here as well? Can there be a transplant operation with thousands of transplanted hashes?

Reply:
Nothing prevents huge transplants :thinking: 

On the other hand, do we really need to expose source hashes in events? If we do not record transplant sources in commit metadata, I'm not sure publishing them in events is very useful due to lack of permanent "source of truth". In the end a transplant is just a regular commit. Why would it matter to event consumers what the source commit was? The event does not provide information on conflict resolutions that happened during the transplanting anyway, does it?

Reply:
> On the other hand, do we really need to expose source hashes in events?

We do not, but I need at least one hash (the last one?) so that I can create a source `Branch` object with a non-null hash, remember? ;-)

How about we expose only the last transplanted hash?

Reply:
I believe it's source "reference" (may be a tag).

TBH, I'm not sure having a "source reference" is very meaningful for transplant results.

I think it is used in the API only to allow for reference-based access checks, but once applied, it does not look very meaningful to me. The same (transplanted) hash may be reachable from multiple references, so what's the point of reporting one reference for it in events? Event consumers probably should be attach any logic to the reference name that was used to trigger a transplant, only to the outcome of such a transplant.

Anyway, I'm fine with reporting only one source hash for transplants.

Reply:
That's a very good point. I will remove the source reference from `TransplantEvent`, that will fix the issue with source hashes as well.

> I believe it's source "reference" (may be a tag).

Yes.

Reply:
Added `TransplantEvent.getSize()` though, could be interesting to inform subscribers about how big was the transplant.